### PR TITLE
Ref clt spacewalk repo sync #3504 ke

### PR DESCRIPTION
**Pull request by keichwa**

### CODE REVIEW

File: modules/reference/pages/command-line-tools.adoc [hunk 1]
```diff
@@ -695,6 +695,68 @@ Show help message and exit.
 
 
 
+[[spacewalk-repo-sync]]
+== Synchronize yum Repositories into {productname} Channels (spacewalk-repo-sync)
```
Comment (mcalmer) [inline modules/reference/pages/command-line-tools.adoc]: yum => software

repomd Repositories would be correct, but this tool can also mirror debian repos. So let's just be generic and say "software repositories"

File: modules/reference/pages/command-line-tools.adoc [hunk 2]
```diff
@@ -695,6 +695,68 @@ Show help message and exit.
...
+
+[command]``spacewalk-repo-sync`` is an important tool on {productname}, which works most of the time in the background.
+With [command]``spacewalk-repo-sync`` you mainly synchronize yum repositories into {productname} channels.
```
Comment (mcalmer) [inline modules/reference/pages/command-line-tools.adoc]: yum => software and maybe we do not need the "mainly".

File: modules/reference/pages/command-line-tools.adoc [hunk 3]
```diff
@@ -695,6 +695,68 @@ Show help message and exit.
...
+
+Use the [option]``--force-all-errata`` to force re-importing all the patches (errata).
+To speed it up you can combine it with the [option]``--no-packages`` option.
+
+To find the root cause of synchronization problems it often helps to view the HTTP log when command]``spacewalk-repo-sync`` is running.
+
+. Set and export the [systemitem]``URLGRABBER_DEBUG``:
++
+----
+export URLGRABBER_DEBUG=DEBUG
```
Comment (mcalmer) [inline modules/reference/pages/command-line-tools.adoc]: Hmm, I fear after the usage of "zypper" this might not work anymore or will give an incomplete result.

@meaksh How would we debug today? The goal is to get the full HTTP protocol conversation printed to the logs.

For zypper you can set the following environment variable:

    ZYPP_MEDIA_CURL_DEBUG=<1|2> Log http headers, if 2 also log server responses.

But we do not use zypper exclusive. For the Downloader part would the URLGRABBER_DEBUG option still work?
Also with python3 ?

File: modules/reference/pages/command-line-tools.adoc (hunk 1 above)
Comment (keichwa) [inline modules/reference/pages/command-line-tools.adoc]:
```suggestion
== Synchronize Software Repositories into {productname} Channels (spacewalk-repo-sync)
```

Comment (keichwa) [reply modules/reference/pages/command-line-tools.adoc]: great, much better :)

File: modules/reference/pages/command-line-tools.adoc (hunk 2 above)
Comment (keichwa) [inline modules/reference/pages/command-line-tools.adoc]:
```suggestion
With [command]``spacewalk-repo-sync`` you synchronize software repositories into {productname} channels.
```

File: modules/reference/pages/command-line-tools.adoc [hunk 4]
```diff
@@ -695,6 +695,68 @@ Show help message and exit.
 
 
 
+[[spacewalk-repo-sync]]
+== Synchronize Software Repositories into {productname} Channels (spacewalk-repo-sync)
```
Comment (Loquacity) [inline modules/reference/pages/command-line-tools.adoc]: Wow, long title. What about:

```suggestion
== Synchronize Software Repositories into Channels
```

Or: 

```suggestion
== Synchronize Repositories  with spacewalk-repo-sync
```

File: modules/reference/pages/command-line-tools.adoc [hunk 5]
```diff
@@ -695,6 +695,68 @@ Show help message and exit.
...
+
+[command]``spacewalk-repo-sync`` is an important tool on {productname}, which works most of the time in the background.
```
Comment (Loquacity) [inline modules/reference/pages/command-line-tools.adoc]:
```suggestion
The [command]``spacewalk-repo-sync`` tool synchronizes software repositories into {productname} channels.
```

File: modules/reference/pages/command-line-tools.adoc [hunk 6]
```diff
@@ -695,6 +695,68 @@ Show help message and exit.
...
+With [command]``spacewalk-repo-sync`` you synchronize software repositories into {productname} channels.
```
Comment (Loquacity) [inline modules/reference/pages/command-line-tools.adoc]:
```suggestion
This usually happens automatically, but you can run it manually if required.
```

File: modules/reference/pages/command-line-tools.adoc [hunk 7]
```diff
@@ -695,6 +695,68 @@ Show help message and exit.
...
+
```
Comment (Loquacity) [inline modules/reference/pages/command-line-tools.adoc]:
```suggestion
```

File: modules/reference/pages/command-line-tools.adoc [hunk 8]
```diff
@@ -695,6 +695,68 @@ Show help message and exit.
...
+But for debugging or solving smaller issues it is useful to run it directly with special parameters.
```
Comment (Loquacity) [inline modules/reference/pages/command-line-tools.adoc]:
```suggestion
This can be useful for debugging or for solving some synchronization problems.
```

Comment (Loquacity) [reply modules/reference/pages/command-line-tools.adoc]: Notes on edits to these three lines:
* The first line should explain what the tool does
* The second line then goes on to explain the circumstances in which you would use it.

It's the same principle as sentence structure, but on a higher level: the first part is the most important, the second part is the lens.

File: modules/reference/pages/command-line-tools.adoc [hunk 9]
```diff
@@ -695,6 +695,68 @@ Show help message and exit.
...
+[command]``spacewalk-repo-sync --list``::
+List all custom channels and the repositories assigned to them.
+[command]``spacewalk-repo-sync  --channel <custom-channel>``::
+Synchronize a single channel ``<custom-channel>`` to all repositories assigned to it with the {webui} or the API.
+
+
+
+=== Solve Checksum Problems
+
+Use the [option]``--deep-verify`` option to ignore cached package checksums, and thus solving checksum problems.
```
Comment (Loquacity) [inline modules/reference/pages/command-line-tools.adoc]:
```suggestion
Use the [option]``--deep-verify`` option to ignore cached package checksums.
This can help with solving checksum problems.
```

File: modules/reference/pages/command-line-tools.adoc [hunk 10]
```diff
@@ -695,6 +695,68 @@ Show help message and exit.
...
+
+
+
+=== Force Re-import Patches (Errata)
```
Comment (Loquacity) [inline modules/reference/pages/command-line-tools.adoc]:
```suggestion
=== Force Re-import Patches
```

File: modules/reference/pages/command-line-tools.adoc [hunk 11]
```diff
@@ -695,6 +695,68 @@ Show help message and exit.
...
+
+Use the [option]``--force-all-errata`` to force re-importing all the patches (errata).
```
Comment (Loquacity) [inline modules/reference/pages/command-line-tools.adoc]:
```suggestion
Use the [option]``--force-all-errata`` option to force re-importing all the patches.
```
I feel like we would need to go into an exposition here about the use of the term "errata", I don't expect it's in common use outside of Red Hat.

File: modules/reference/pages/command-line-tools.adoc [hunk 12]
```diff
@@ -695,6 +695,68 @@ Show help message and exit.
...
+To speed it up you can combine it with the [option]``--no-packages`` option.
```
Comment (Loquacity) [inline modules/reference/pages/command-line-tools.adoc]:
```suggestion
To make this command run faster, you can use the [option]``--no-packages`` option.
This option excludes packages from the operation. 
```

File: modules/reference/pages/command-line-tools.adoc [hunk 13]
```diff
@@ -695,6 +695,68 @@ Show help message and exit.
...
+
+To find the root cause of synchronization problems it often helps to view the HTTP log when command]``spacewalk-repo-sync`` is running.
```
Comment (Loquacity) [inline modules/reference/pages/command-line-tools.adoc]:
```suggestion
To find the root cause of synchronization problems you can look at the HTTP log as [command]``spacewalk-repo-sync`` is running.
```

File: modules/reference/pages/command-line-tools.adoc [hunk 14]
```diff
@@ -695,6 +695,68 @@ Show help message and exit.
...
+
+. Set and export the [systemitem]``URLGRABBER_DEBUG``:
++
+----
+export URLGRABBER_DEBUG=DEBUG
+----
+. Run the synchronization command:
```
Comment (Loquacity) [inline modules/reference/pages/command-line-tools.adoc]:
```suggestion
. Start the synchronization:
```

File: modules/reference/pages/command-line-tools.adoc [hunk 15]
```diff
@@ -695,6 +695,68 @@ Show help message and exit.
...
++
+
+----
+/usr/bin/spacewalk-repo-sync --channel <channel-label> --type yum
+----
+
+To increase the debug level, add the [option]``-vvv`` option.
+
+When debugging is done, disable the debug mode:
```
Comment (Loquacity) [inline modules/reference/pages/command-line-tools.adoc]:
```suggestion
When debugging is finished, disable debug mode:
```

File: modules/reference/pages/command-line-tools.adoc [hunk 16]
```diff
@@ -695,6 +695,68 @@ Show help message and exit.
...
+
+----
+unset URLGRABBER_DEBUG
+----
+
+
+
+=== For More Information
+
+For a complete list of command line option, see the [command]``spacewalk-repo-sync`` manpage ([command]``man spacewalk-repo-sync``).
```
Comment (Loquacity) [inline modules/reference/pages/command-line-tools.adoc]:
```suggestion
For a complete list of command line options, see the [command]``spacewalk-repo-sync`` manpage:

 ----
 man spacewalk-repo-sync
 ----
```

File: modules/reference/pages/command-line-tools.adoc [hunk 17]
```diff
@@ -695,6 +695,68 @@ Show help message and exit.
...
+
+
+
 [[syncing.suse.mgr.repositories.scc]]
 == Syncing SUSE Manager Repositories from SCC (mgr-sync)
```
Comment (Loquacity) [inline modules/reference/pages/command-line-tools.adoc]:
```suggestion
== Synchronize SUSE Manager Repositories from SCC (mgr-sync)
```

Ooh, drive-by edits! :wink:

File: modules/reference/pages/command-line-tools.adoc (hunk 4 above)
Comment (keichwa) [reply modules/reference/pages/command-line-tools.adoc]: I'll take the or version (because it is in the Ref Guide, I'd like to see the name of the tool we are discussing in the title.

File: modules/reference/pages/command-line-tools.adoc (hunk 10 above)
Comment (keichwa) [reply modules/reference/pages/command-line-tools.adoc]: To help users coming from satellite, we used to add "errata" when talking about patches.  Maybe, this is no longer useful and causes more trouble than good.  I do not want to decide about it.

File: modules/reference/pages/command-line-tools.adoc (hunk 3 above)
Comment (meaksh) [reply modules/reference/pages/command-line-tools.adoc]: There are two things here:

Setting `ZYPP_MEDIA_CURL_DEBUG=<1|2>` as env variable would work for the metadata downloading part made by Zypper. So, i.a., `ZYPP_MEDIA_CURL_DEBUG=2 spacewalk-repo-sync -c MYCHANNEL` will log the HTTP conversation into `/var/log/zypper.log`.

For the RPM downloading part, we would need to keep using `URLGRABBER_DEBUG` (since the backend for downloading the RPM is still `urlgrabber`), but I just tested it now and it seems there is an issue when running on Python3.

I've just filed a bug for this: https://bugzilla.suse.com/show_bug.cgi?id=1152514

Comment (keichwa) [reply modules/reference/pages/command-line-tools.adoc]: @meaksh thanks for checking and feedback!

**Comment by keichwa** [comment]
thanks for all your help!
